gupb/model/pandas_profiler.py

```diff
@@ -0,0 +1,164 @@
+from time import time
+from datetime import datetime
+from dataclasses import dataclass
+from typing import Optional
+from types import ModuleType, FunctionType
+from .characters import Action
+from os import makedirs, path
+from gc import get_referents
+
+import sys
+
+BLACKLIST = type, ModuleType, FunctionType
+
+
+def getsize(obj):
+    """sum size of object & members."""
+    if isinstance(obj, BLACKLIST):
+        raise TypeError("getsize() does not take argument of type: " + str(type(obj)))
+    seen_ids = set()
+    size = 0
+    objects = [obj]
+    while objects:
+        need_referents = []
+        for obj in objects:
+            if not isinstance(obj, BLACKLIST) and id(obj) not in seen_ids:
+                seen_ids.add(id(obj))
+                size += sys.getsizeof(obj)
+                need_referents.append(obj)
+        objects = get_referents(*need_referents)
+    return size
+
+
+@dataclass
+class DecisionRecord:
+    epoch: int
+    step: int
+    controller: str
+    decision_time: float
+    memory_before: float
+    memory_after: float
+    action: Optional[Action]
+    exception: Optional[Exception]
+
+    def __str__(self):
+        return (
+            f"{self.epoch},{self.step},{self.controller},"
+            + f"{self.decision_time},{self.memory_before},{self.memory_after},"
+            + f"{self.action.name if self.action is not None else ''},"
+            + f"{repr(self.exception) if self.exception is not None else ''}\n"
+        )
+
+    @staticmethod
+    def header():
+        return (
+            "epoch,step,controller_name,decision_time,memory_before,memory_after,"
+            + "action_name,exception_returned\n"
```

Comment:
To mogłoby powstawać dynamicznie w oparciu o nazwy pól w dataclass, tak samo metoda __str__ - obecnie jest sporo powtórzonego kodu.